### PR TITLE
Normalize HTTPD and APACHE

### PR DESCRIPTION
**Pull request by chernjie**

### CODE REVIEW

**Comment by chernjie** [comment]
@purbon it should not break compatibility now

File: patterns/httpd [hunk 1]
```diff
@@ -0,0 +1,15 @@
+HTTPDUSER %{EMAILADDRESS}|%{USER}
+HTTPDERROR_DATE %{DAY} %{MONTH} %{MONTHDAY} %{TIME} %{YEAR}
+
+# Log formats
+HTTPD_COMMONLOG %{IPORHOST:clientip} %{HTTPDUSER:ident} %{USER:auth} \[%{HTTPDATE:timestamp}\] "(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})" %{NUMBER:response} (?:%{NUMBER:bytes}|-)
```
Comment (jordansissel) [inline patterns/httpd]: Oh lovely!

While we're here, any thoughts on making `bytes` and `httpversion` both numbers? (int and float, in grok, respectively)

File: spec/patterns/httpd_spec.rb [hunk 2]
```diff
@@ -2,23 +2,112 @@
 require "spec_helper"
 require "logstash/patterns/core"
 
+describe "HTTPD_COMBINEDLOG" do
+
+  let(:value) { '83.149.9.216 - - [24/Feb/2015:23:13:42 +0000] "GET /presentations/logstash-monitorama-2013/images/kibana-search.png HTTP/1.1" 200 203023 "http://semicomplete.com/presentations/logstash-monitorama-2013/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/32.0.1700.77 Safari/537.36"'}
+
+  it "generates the clientip field" do
+    expect(grok_match(subject, value)).to include(
+      'clientip' => '83.149.9.216',
```
Comment (jordansissel) [inline spec/patterns/httpd_spec.rb]: Whoa, this works? Amazing! I love rspec so much. Thanks for teaching me this :)

File: patterns/httpd (hunk 1 above)
Comment (jordansissel) [inline patterns/httpd]: If we do change, for example, to use `%{NUMBER:bytes:int}` we'll want the specs to expect numbers, not strings.

For background, I've wanted bytes (especially) and httpversion stored as numbers in Elasticsearch for years, but I didn't want to break compat. This PR is a perfect opportunity to do that.

Thoughts?

Comment (chernjie) [inline patterns/httpd]: it makes sense to make bytes `int`, then you can plot bandwidth in Kibana.
As for `httpversion`, care to enlighten me what are the benefits?

Comment (purbon) [inline patterns/httpd]: @jordansissel I think we should convert more of this strings to integers,  see #83, I think it makes sense to to do it everywhere that makes sense in the patterns. What do you think?

File: spec/patterns/httpd_spec.rb (hunk 2 above)
Comment (purbon) [inline spec/patterns/httpd_spec.rb]: ++ This works and is super nice! More intel: https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/include-matcher

**Comment by chernjie** [comment]
hi @jordansissel  and @purbon , any updates for this PR?

File: patterns/httpd [hunk 3]
```diff
@@ -0,0 +1,15 @@
+HTTPDUSER %{EMAILADDRESS}|%{USER}
+HTTPDERROR_DATE %{DAY} %{MONTH} %{MONTHDAY} %{TIME} %{YEAR}
+
+# Log formats
+HTTPD_COMMONLOG %{IPORHOST:clientip} %{HTTPDUSER:ident} %{USER:auth} \[%{HTTPDATE:timestamp}\] "(?:%{WORD:verb} %{NOTSPACE:request}(?: HTTP/%{NUMBER:httpversion})?|%{DATA:rawrequest})" %{NUMBER:response} (?:%{NUMBER:bytes:int}|-)
```
Comment (jordansissel) [inline patterns/httpd]: Should httpversion be `:float` also? Or no?

Comment (chernjie) [inline patterns/httpd]: well, no harm future-proofing. As of today, I only know of `1.0` and `1.1` in the wild. `0.9` is merged into `1.0` and `2.0` is not commonly used yet.

Comment (chernjie) [inline patterns/httpd]: Do you want `response` to be `%{NUMBER:response:int}`

Comment (chernjie) [inline patterns/httpd]: @jordansissel any updates on this?

**Comment by jsvd** [comment]
jenkins, test it

**Comment by joshuaspence** [comment]
+1 on this change.

**Comment by jsvd** [comment]
@chernjie could you rebase this against master so I can merge it? thanks!

**Comment by chernjie** [comment]
@jsvd rebased. Should I separate the "breaking-compatibility" commits into a separate pull request?

**Comment by jsvd** [comment]
Does it still break compatibility? seems to maintain all previous patterns

**Comment by chernjie** [comment]
the last 3 commits break compatibility, changing:
- bytes to `int`
- httpversion to `float`
- response to `int`

**Comment by chernjie** [comment]
I think it would be better if the last 3 changes are moved into a separate PR

**Comment by jsvd** [comment]
+1

**Comment by chernjie** [comment]
@jsvd last three commits have been moved to #174

**Comment by jsvd** [comment]
LGTM

**Comment by jsvd** [comment]
great work @chernjie! thanks for the patience.

**Pull request closed without merging**